js/main.js

```diff
@@ -80,11 +83,82 @@ function searchBar(event) {
     var $weaponToBeHidden = $allWeapons[nameIndex].closest('.green-card');
-    $weaponToBeHidden.classList.add('display-hidden');
+    $weaponToBeHidden.classList.add('hidden');
     if (weaponName.includes(inputValue)) {
-      $weaponToBeHidden.classList.remove('display-hidden');
+      $weaponToBeHidden.classList.remove('hidden');
     }
   }
 }
 
 $form.addEventListener('submit', searchBar);
+
+var $imageInDetails = document.querySelector('.image-in-details');
+var $descriptionInDetails = document.querySelector('.description');
+var $commonLocationsInDetails = document.querySelector('.common-locations');
+function bringUserToDetailsPage(event) {
+  if (event.target.matches('.weapons')) {
+    var weaponThatWasClicked = event.target.closest('.green-card');
+    var weaponThatWasClickedID = weaponThatWasClicked.getAttribute('id');
+    weaponThatWasClickedID = parseInt(weaponThatWasClickedID);
+    var dataBOTW = xhr.response.data;
+    for (var dataIndex = 0; dataIndex < dataBOTW.length; dataIndex++) {
+      if (dataBOTW[dataIndex].id === weaponThatWasClickedID) {
+        data.clicked.name = dataBOTW[dataIndex].name;
+        data.clicked.image = dataBOTW[dataIndex].image;
+        data.clicked.description = dataBOTW[dataIndex].description;
+        data.clicked.common_locations = dataBOTW[dataIndex].common_locations;
+        data.clicked.id = dataBOTW[dataIndex].id;
+        $imageInDetails.setAttribute('src', data.clicked.image);
+        $descriptionInDetails.textContent = data.clicked.description;
+        $commonLocationsInDetails.textContent = data.clicked.common_locations;
+      }
+    }
+  }
+
+}
+
+document.addEventListener('click', bringUserToDetailsPage);
+
+var $allViews = document.querySelectorAll('.view');
+function switchViews(event) {
+  if (event.target.matches('.weapons')) {
+    for (var viewIndex = 0; viewIndex < $allViews.length; viewIndex++) {
+      if ($allViews[viewIndex].getAttribute('data-view') === 'home-page') {
+        $allViews[viewIndex].classList.replace('display', 'hidden');
+      }
+      if ($allViews[viewIndex].getAttribute('data-view') === 'details') {
+        $allViews[viewIndex].classList.replace('hidden', 'display');
+        data.view = 'details';
+      }
+    }
+  }
+  if (event.target.matches('.search-icon')) {
+    for (viewIndex = 0; viewIndex < $allViews.length; viewIndex++) {
+      if ($allViews[viewIndex].getAttribute('data-view') === 'home-page') {
+        $allViews[viewIndex].classList.replace('hidden', 'display');
+      }
+      if ($allViews[viewIndex].getAttribute('data-view') === 'details') {
+        $allViews[viewIndex].classList.replace('display', 'hidden');
+        data.view = 'home-page';
+        clearDataClicked();
+      }
+    }
+  }
+}
+document.addEventListener('click', switchViews);
+
+function clearDataClicked() {
+  data.clicked = {};
+}
+
+function previousDataView(data) {
+  for (var viewIndex = 0; viewIndex < $allViews.length; viewIndex++) {
```

Comment:
You should make a function that receives the name of the view to go to and shows that view, so that you don't have to repeat this looping over a list every time you want to change views separate from an event handler.